### ISSUE TITLE
Bun doesn't alway respect bunfig.toml registry configuration

### ISSUE THREAD

**Issue by ahmedelgabri** [oven-sh/bun#4901]
### What version of Bun is running?

1.0.0

### What platform is your computer?

Darwin 22.6.0 arm64 arm

### What steps can reproduce the bug?

For this, you will need two projects, one using a different registry than `https://registry.npmjs.org`

1. Go to project one
2. Add `bunfig.toml` file in the root of the project with the following

```toml
[install]
registry = { url = "https://some-custom-registry", token = "$MY_TOKEN" }
```

3. run `bun install`
4. Now go to project two
5. run `bun install` 
6. It should fail to try to resolve _some_ dependencies from `https://some-custom-registry` instead of `https://registry.npmjs.org`. 
```
GET https://some-custom-registry/-/merge-descriptors-1.0.1.tgz - 401
  🔍 Resolving...
```

Adding `bunfig.toml` file to the root of the project or even `~/.bunfig.toml` and explicitly setting the registry to `https://registry.npmjs.org` doesn't help & I also couldn't find a way to clear the bun cache etc… so I'm currently stuck with this and can't use `bun install` at all on any project.

### What is the expected behavior?

To not error out and to pull in packages from `https://registry.npmjs.org` by default but at least to respect `bunfig.toml` file if registry is set.

### What do you see instead?

401 errors like this 

```
GET https://some-custom-registry/-/merge-descriptors-1.0.1.tgz - 401
  🔍 Resolving...
```

**Comment by iwt-jandwinger** [comment]
related to  https://github.com/oven-sh/bun/issues/4666

**Comment by ahmedelgabri** [comment]
> related to  https://github.com/oven-sh/bun/issues/4666

@iwt-jandwinger I don't think they are related.

**Comment by inimicus** [comment]
As an additional repro:

Doesn't work when in project root, works in `$HOME`:
```toml
[install.scopes]
"@fortawesome" = { url = "https://npm.fontawesome.com/", token = "XXX" }
```

**Comment by ahmedelgabri** [comment]
@inimicus I didn't try scopes. But in my case, if I use a private registry in one repo, any other repo will try to use this registry to pull _some_ dependencies. My assumption here is that it's trying to use the private registry for the cached dependencies that it installed before and this is why it fails with 401 because these repos don't use a private registry., but I can't properly debug this to validate this assumption.

**Comment by tutturen** [comment]
I am experiencing the same problem.

**Comment by inimicus** [comment]
After revisiting this, my repro mentioned above is an entirely different issue. I've opened a separate enhancement/feature issue [here](https://github.com/oven-sh/bun/issues/5936) related what I encountered.

**Comment by ahmedelgabri** [comment]
> @inimicus I didn't try scopes. But in my case, if I use a private registry in one repo, any other repo will try to use this registry to pull _some_ dependencies. My assumption here is that it's trying to use the private registry for the cached dependencies that it installed before and this is why it fails with 401 because these repos don't use a private registry., but I can't properly debug this to validate this assumption.

My assumption here seems to be correct, I just tested again after the [`1.1`](https://bun.sh/blog/bun-v1.1) release, the same issue is happening still. But when I run `bun install --no-cache` it works fine.

**Comment by dylan-conway** [comment]
Fixed in #11606 and will be included in v1.1.13. Note that `bun.lockb` will need to be deleted and regenerated. If you continue have problems please reopen this issue or open a new one.
